akka-persistence/src/test/scala/akka/persistence/PersistentActorSpec.scala

```diff
@@ -1213,10 +1213,8 @@ abstract class PersistentActorSpec(config: Config) extends PersistenceSpec(confi
       val persistentActor = asyncPersistPersistentActor
       persistentActor ! Cmd("x")
       persistentActor ! Cmd("y")
-      expectMsg("x")
-      expectMsg("y") // "y" command was processed before event persisted
-      expectMsg("x-1")
-      expectMsg("y-2")
+      val messages = receiveN(4)
+      messages.toSet should equal(Set("x", "y", "x-1", "y-2"))
```

Comment:
While this will surely make the test more stable, also it doesn't test what it was intended to test anymore ;).

The test was intended to check that the actor can start processing the command "y" (evidenced by the reply "y") before the persist of the command "x" has completed. With this change, the test will also succeed when then implementation no longer allows that.

If the 'persist' operation is fast, however, it is entirely possible that the event handling would finish before the processing of "y", so seeing x-1 sooner is not 'wrong' per se.

While we should not over-use it, perhaps an `awaitAssert`/`eventually` around this whole test might be justifiable?

Reply:
> If the 'persist' operation is fast, however, it is entirely possible that the event handling would finish before the processing of "y", so seeing x-1 sooner is not 'wrong' per se.

I think observing `Set("x", "x-1", "y", "y-2")` in the test is valid output but it's not proving the feature is correct. `eventually` sounds like a good plan to me.

Reply:
> I think observing `Set("x", "x-1", "y", "y-2")` in the test is valid output but it's not proving the feature is correct.

completely agree!